processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -375,14 +375,16 @@ public Type erasure() {
      *
      * @return {@code true} if and only if this type is assignable to the given other type.
      */
-    // TODO This doesn't yet take wild card types into account; e.g. ? extends Integer wouldn't be assignable to Number
-    // atm.
     public boolean isAssignableTo(Type other) {
         if ( equals( other ) ) {
             return true;
         }
 
-        return typeUtils.isAssignable( typeMirror, other.typeMirror );
+        TypeMirror typeMirrotToMatch = isWildCardExtendsBound() ? getTypeBound().typeMirror : typeMirror;
```

Comment:
There is a type in the variable `typeMirrotToMatch`, it should be `typeMirrorToMatch` the same for the `otherTypeMirrotToMatch`, it should be `otherTypeMirrorToMatch`.